### PR TITLE
get tfp.distribution.Categorical probs using public interface

### PR DESCRIPTION
**Pull request by witwolf**
avoid visit private members  mentioned by @emailweixu  at  #143

### CODE REVIEW

**Comment by emailweixu** [comment]
Is there any papers about the discrete SAC? For discrete case, because we already know the Q values, instead of using an actor network, it seems that we can have an analytical result for the action distribution by minimizing  E_{a~p(s)} (\alpha log p(a) - Q(a))

File: alf/algorithms/sac_algorithm.py [hunk 1]
```diff
@@ -240,7 +240,7 @@ def actor_loss_fn(dqda, action):
                 step_type=exp.step_type,
                 network_state=state.critic2)
 
-            action_probs = action_distribution._probs
+            action_probs = action_distribution.probs
```
[the remark this answers is not in the document]
Comment (witwolf) [reply alf/algorithms/sac_algorithm.py]: added

**Comment by witwolf** [comment]
i did not found papers about discrete SAC but got mentions:    
https://github.com/haarnoja/sac/issues/21#issuecomment-437906867
[spiningup sac doc](https://github.com/openai/spinningup/blob/master/docs/algorithms/sac.rst)
and implementations:
[SAC_Discrete](https://github.com/p-christ/Deep-Reinforcement-Learning-Algorithms-with-PyTorch/blob/master/agents/actor_critic_agents/SAC_Discrete.py)

**Comment by emailweixu** [comment]
> Is there any papers about the discrete SAC? For discrete case, because we already know the Q values, instead of using an actor network, it seems that we can have an analytical result for the action distribution by minimizing E_{a~p(s)} (\alpha log p(a) - Q(a))

Actually, minimizing E_{a~p(a)} (\alpha log p(a) - Q(a)), we can get p(a) = 1/ Z  exp(Q(a)/alpha), thus no need to use a separate actor network.

**Comment by witwolf** [comment]
> Actually, minimizing E_{a~p(a)} (\alpha log p(a) - Q(a)), we can get p(a) = 1/ Z exp(Q(a)/alpha), thus no need to use a separate actor network.

yeah , it's no need to use a separate actor network in this way

**Pull request merged**